### PR TITLE
Package for SPM8 (implemented for octave) 

### PR DESCRIPTION
**Pull request by TheChymera**
SPM is the most popular softrware package for analyzing various types of brain imaging data. It is an invaluable addition to any scientific software repository. SPM8 (released as a live package) is the current stable version, and even though the software package was written for MATLAB most of its functions can also be used with octave (more on that here: http://en.wikibooks.org/wiki/SPM/Octave ).

### CODE REVIEW

File: sci-biology/spm/spm-8.ebuild [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Copyright 1999-2013 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=5
+
+DESCRIPTION="Analysis of brain imaging data sequences for Octave or Matlab"
+HOMEPAGE="http://www.fil.ion.ucl.ac.uk/spm/"
+URI_BASE="http://www.fil.ion.ucl.ac.uk/spm/download/restricted/idyll/"
+SRC_URI="${URI_BASE}${PN}${PV}.zip"
```
Comment (junghans) [inline sci-biology/spm/spm-8.ebuild]: `${URI_BASE}` is only used once, so please inline it into `SRC_URI`.

File: sci-biology/spm/spm-8.ebuild [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+LICENSE="GPL-2+"
+SLOT="8"
```
Comment (junghans) [inline sci-biology/spm/spm-8.ebuild]: Why? I think `SLOT="0"` would be more natural.

**Comment by junghans** [comment]
Sorry for the delay!
1. keep it!
2. see my comment
5. please run `repoman commit` instead of `git commit` to commit the final ebuild
7. I will merge it.

File: sci-biology/spm/spm-8.ebuild (hunk 2 above)
Comment (TheChymera) [inline sci-biology/spm/spm-8.ebuild]: SPM has a weird versioning scheme, whereby different versions are (almost) treated as different programs. For instance, the version number is part of the executable name. I thought slots would be a good Gentoo analogue for that, and would also allow users to have multiple versions of it installed. Most work is still done on SPM-8, and users would need it to reproduce past analyses, but newer or experimental projects would likely be run on SPM-12.

**Comment by TheChymera** [comment]
@junghans thanks for getting back to this! How do I rebase it with the current master?

File: sci-biology/spm/spm-8.ebuild (hunk 2 above)
Comment (marbre) [inline sci-biology/spm/spm-8.ebuild]: I think what @junghans suggests is using `SLOT="0"` for spm8 and `SLOT="1"` for spm12.

**Comment by marbre** [comment]
Set a remote (upstream) https://help.github.com/articles/configuring-a-remote-for-a-fork/
Afterwards do
> git fetch upstream
> git rebase upstream/master

**Comment by TheChymera** [comment]
It fails, and I don't know how to fix it:

```
chymera@zenbookhost ~/src/sci $ git branch
  fsl
  master
  neurogentoo
  patch-1
* spm-octave
chymera@zenbookhost ~/src/sci $ git fetch official
remote: Counting objects: 1338, done.
remote: Compressing objects: 100% (880/880), done.
remote: Total 1338 (delta 590), reused 1119 (delta 450)
Receiving objects: 100% (1338/1338), 529.82 KiB | 0 bytes/s, done.
Resolving deltas: 100% (590/590), done.
From https://github.com/gentoo-science/sci
 * [new branch]      alternatives -> official/alternatives
 * [new branch]      ccp4-640   -> official/ccp4-640
 * [new branch]      int64-build-eclass -> official/int64-build-eclass
 * [new branch]      jlec       -> official/jlec
   f66fc75..44a2983  master     -> official/master
 * [new branch]      mpi        -> official/mpi
chymera@zenbookhost ~/src/sci $ git rebase official/master
First, rewinding head to replay your work on top of it...
Applying: Initial spm8 ebuild, doesn't work atm.
Applying: Set spm8 instead of spm8-9999 as the working dir
Applying: Used emake instead of make, specified distdir explicitly in the make commands.
Applying: changed cd to src (directory structure was messed uo)
Applying: added -j1 option to the second emake call (otherwise some file doesn't get found)
Applying: initial commit
Applying: added separate src_prepare and src_compile sections
Applying: added tcsh dependency
Applying: Added cp call to move everything to some final dir
Applying: Added proper print call instead of pwd to see cp target location
Applying: Typo
Applying: Added dodir and improved mv call, switched to opt instead of usr/src
Applying: newline instead of ";" and no -r option for mv
Applying: removed static lib tag for motif, also removed second sed line
Applying: proper handling of live ebuild source
Applying: Added aditional deps as suggested in this thread http://afni.nimh.nih.gov/afni/community/board/read.php?1,85348,85348#msg-85348
Applying: It finally works!
Applying: Renamed main Makefile so that the ptaylor Makefile plays along.
Applying: Added ewarn call to inform the user on adding SPM8 to the octave path
Applying: simplified emake call.
Applying: Added -j1 to emake (apparently multithreading may screw some eispack/ things up)
Applying: replaced URI_BASENAMe with PN, directory now just PN
Applying: using emake -C src instead of cs on every section
Applying: Sourced new Makefile
Applying: cosmetic changes
Applying: Replaced the standard src_install function with a mv call.
Applying: Added doenvd file /etc/env.d/99afni, now just mv-ing the build dorectory (linux_fedora...) in the src_install() phase
Applying: Added metadata
Applying: Removed AFMI from this branch
Applying: new license, slots, versioning, destination directory for *.m files
Applying: removed “spm8”
Applying: remove src dir after compile, echo -v for MEXOPTS
Applying: Removed interit unpacker
Applying: Removed rm call for .../src/
Applying: sci-biology/spm: switched order of RDEPEND and DEPEND; using SRC_URI
Applying: Minor improvements + added metadata.xml
Using index info to reconstruct a base tree...
M	sci-biology/spm/spm-8.ebuild
Falling back to patching base and 3-way merge...
Auto-merging sci-biology/spm/spm-8.ebuild
CONFLICT (content): Merge conflict in sci-biology/spm/spm-8.ebuild
Failed to merge in the changes.
Patch failed at 0036 Minor improvements + added metadata.xml
The copy of the patch that failed is found in:
   /home/chymera/src/sci/.git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

chymera@zenbookhost ~/src/sci $ cat /home/chymera/src/sci/.git/rebase-apply/patch
68ebba15191cddb1fc39aba9ddb66f88f38530a7
diff --git a/sci-biology/spm/spm-8.ebuild b/sci-biology/spm/spm-8.ebuild
index dd29362e7baa6252bbd85466896d402da6635957..41b3ca79ba6b6d193141e3a726a8aea308b4ab94 100644
--- a/sci-biology/spm/spm-8.ebuild
+++ b/sci-biology/spm/spm-8.ebuild
@@ -11,13 +11,13 @@ SRC_URI="" # SRC_URI is left blank on live ebuild
 
 LICENSE="GPL-2+"
 SLOT="8"
-KEYWORDS=""
+KEYWORDS="~amd64"
 
 DEPEND="${RDEPEND}
+	app-arch/unzip
 	"
 
-RDEPEND=">=sci-mathematics/octave-3.6.4
-	"
+RDEPEND=">=sci-mathematics/octave-3.6.4"
 
 S=${WORKDIR}/${PN}${PV}
 
```

**Comment by TheChymera** [comment]
@junghans -looking good now?

**Comment by marbre** [comment]
@TheChymera https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/ ... I recommend to get more familiar with the documentation.

You're ebuild is still `SLOT="8"`...

**Comment by TheChymera** [comment]
Does everybody agree that this is a bad idea? As I explained earlier, users may want spm-8 alongside spm-12. And slots seem like a good way to do this.

**Comment by junghans** [comment]
It makes sense as long as
* there are no file conflicts between the different slot
* packages depend on a specific slot (or `:=`) of `fsl`

**Comment by TheChymera** [comment]
1) Currently we only have one slot, so no package collisions there.
2) so I should edit the nipype ebuild then?

**Comment by junghans** [comment]
1) Let me rephrase if you plan to have multiple version installed simultaneously, make sure they would have file conflicts. Otherwise multiple SLOTs would be useless.
2) You can, but let's get his pull request merge first.
3) I guess you rebased the wrong way around, you could try
```
git checkout spm-octave
git reset --hard 30fc751
git pull --rebase git@github.com:gentoo-science/sci.git master
```

**Comment by TheChymera** [comment]
@junghans 
1) I think the best way to do that is just to place the files in different folders, as per `insinto $(octave-config --m-site-dir)` stuff is just dumped in some octave `.m` file dir? 
3) apparently this rebasing thing doesn't work - what on earth is wrong? I ran exactly what you told me...

**Comment by junghans** [comment]
1) Or install it under different file names. Of course the question is, what do multiple installs of  `spm` get you?
3) Apparently, without any output, I have no idea what didn't work! Another try:
```
git checkout spm-octave
git reset --hard 30fc751
```
now make sure you are on right branch (check output of `git branch`) and the last commit is the `spm` commit from Oct, 13 2013 (check output of `git log -1`). Now instead of rebasing try to merge:
```
git pull git@github.com:gentoo-science/sci.git master
```
(you might have to resolve conflict and commit again). Check the branch again (`git branch`). Now you have to push the new branch head to your repository:
```
git push -f git@github.com:TheChymera/gentoo-overlay.git spm-octave:spm-octave
```
Good luck!

**Comment by TheChymera** [comment]
3)I think the problem lies right with the first commands. after I do this:
```
chymera@zenbookhost ~/src/sci $ git checkout spm-octave
Switched to branch 'spm-octave'
chymera@zenbookhost ~/src/sci $ git reset --hard 30fc751
HEAD is now at 30fc751 Merge branch 'spm-octave' of https://github.com/TheChymera/neurogentoo into spm-octave
chymera@zenbookhost ~/src/sci $ git branch
  fsl
  master
  neurogentoo
  patch-1
  patch-2
  pythontex-bump
  sci
* spm-octave
```
I get this
```
commit 30fc751ace5889a7e3ceadad5a0d5e067dddf058
Merge: 2c80a4f 078277f
Author: Horea Christian <h.chr@mail.ru>
Date:   Mon Feb 9 15:18:55 2015 +0100

    Merge branch 'spm-octave' of https://github.com/TheChymera/neurogentoo into spm-octave
```
from `git log -1`

**Comment by junghans** [comment]
You basically need use the last good commit (before the merge) there, maybe ` 2c80a4f` is better.
You can use `gitk` to find it.

**Comment by TheChymera** [comment]
It seems to have worked, but now I can no longer commit. What should I try next?

```
chymera@zenbookhost ~/src/neurogentoo $ git reset --hard 2c80a4f
HEAD is now at 2c80a4f inlined URI_BASE
chymera@zenbookhost ~/src/neurogentoo $ git pull official master
remote: Counting objects: 97, done.
remote: Compressing objects: 100% (65/65), done.
remote: Total 97 (delta 34), reused 74 (delta 31)
Unpacking objects: 100% (97/97), done.
From https://github.com/gentoo-science/sci
 * branch            master     -> FETCH_HEAD
   a7c8e86..32a6811  master     -> official/master
Auto-merging dev-util/smartgit/smartgit-6.5.6.ebuild
Removing app-shells/mpibash/mpibash-4.3_p30-r1.ebuild
Removing app-shells/mpibash/metadata.xml
Removing app-shells/mpibash/files/mpibash-4.3.patch
Removing app-shells/mpibash/files/bash-4.3-term-cleanup.patch
Removing app-shells/mpibash/files/bash-4.3-parse-time-keyword.patch
Removing app-shells/mpibash/files/bash-4.3-compat-lvl.patch
Removing app-shells/mpibash/files/bash-4.3-append-process-segfault.patch
Removing app-shells/mpibash/Manifest
Removing app-shells/mpibash/ChangeLog
Merge made by the 'recursive' strategy.
 .travis.yml                                                        |    7 +-
 app-shells/mpibash/ChangeLog                                       |   25 -
 app-shells/mpibash/Manifest                                        |   31 -
 app-shells/mpibash/files/bash-4.3-append-process-segfault.patch    |   18 -
 app-shells/mpibash/files/bash-4.3-compat-lvl.patch                 |   13 -
 app-shells/mpibash/files/bash-4.3-parse-time-keyword.patch         |   30 -
 app-shells/mpibash/files/bash-4.3-term-cleanup.patch               |   47 --
 app-shells/mpibash/files/mpibash-4.3.patch                         | 1565 -------------------------------------
 app-shells/mpibash/metadata.xml                                    |   16 -
 app-shells/mpibash/mpibash-4.3_p30-r1.ebuild                       |  277 -------
 dev-lang/vtune/ChangeLog                                           |    8 +-
 dev-lang/vtune/Manifest                                            |    1 +
 dev-lang/vtune/vtune-15.2.393444.ebuild                            |   50 ++
 dev-util/smartgit/ChangeLog                                        |   21 +-
 dev-util/smartgit/Manifest                                         |    2 +-
 dev-util/smartgit/{smartgit-6.0.6.ebuild => smartgit-6.5.6.ebuild} |   24 +-
 licenses/XPLOR-NIH                                                 |   37 +
 profiles/categories                                                |  127 ++-
 sci-biology/eval/ChangeLog                                         |    5 +-
 sci-biology/eval/eval-2.2.8.ebuild                                 |   15 +-
 sci-biology/primerD/ChangeLog                                      |    5 +-
 sci-biology/primerD/primerD-1.0.0.ebuild                           |   13 +
 sci-chemistry/votca-csg/ChangeLog                                  |    5 +-
 sci-chemistry/votca-csg/votca-csg-9999.ebuild                      |   39 +-
 sci-chemistry/xplor-nih/ChangeLog                                  |    9 +
 sci-chemistry/xplor-nih/Manifest                                   |    2 +
 sci-chemistry/xplor-nih/metadata.xml                               |    8 +
 sci-chemistry/xplor-nih/xplor-nih-2.38.ebuild                      |   49 ++
 sci-libs/libfac/Manifest                                           |    1 -
 sci-libs/mkl/ChangeLog                                             |    7 +-
 sci-libs/mkl/Manifest                                              |    1 +
 sci-libs/mkl/mkl-11.2.2.164.ebuild                                 |  143 ++++
 32 files changed, 527 insertions(+), 2074 deletions(-)
 delete mode 100644 app-shells/mpibash/ChangeLog
 delete mode 100644 app-shells/mpibash/Manifest
 delete mode 100644 app-shells/mpibash/files/bash-4.3-append-process-segfault.patch
 delete mode 100644 app-shells/mpibash/files/bash-4.3-compat-lvl.patch
 delete mode 100644 app-shells/mpibash/files/bash-4.3-parse-time-keyword.patch
 delete mode 100644 app-shells/mpibash/files/bash-4.3-term-cleanup.patch
 delete mode 100644 app-shells/mpibash/files/mpibash-4.3.patch
 delete mode 100644 app-shells/mpibash/metadata.xml
 delete mode 100644 app-shells/mpibash/mpibash-4.3_p30-r1.ebuild
 create mode 100644 dev-lang/vtune/vtune-15.2.393444.ebuild
 rename dev-util/smartgit/{smartgit-6.0.6.ebuild => smartgit-6.5.6.ebuild} (59%)
 create mode 100644 licenses/XPLOR-NIH
 create mode 100644 sci-chemistry/xplor-nih/ChangeLog
 create mode 100644 sci-chemistry/xplor-nih/Manifest
 create mode 100644 sci-chemistry/xplor-nih/metadata.xml
 create mode 100644 sci-chemistry/xplor-nih/xplor-nih-2.38.ebuild
 create mode 100644 sci-libs/mkl/mkl-11.2.2.164.ebuild
chymera@zenbookhost ~/src/neurogentoo $ git commit -a
On branch spm-octave
Your branch and 'origin/spm-octave' have diverged,
and have 24 and 111 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)
Untracked files:
	sci-biology/bowtie/files/
	sci-biology/embassy-cbstools/files/embassy-cbstools-1.0.0-r2.patch
	sci-biology/embassy-domainatrix/files/embassy-domainatrix-0.1.0-r5.patch
	sci-biology/embassy-domalign/files/embassy-domalign-0.1.0-r5.patch
	sci-biology/embassy-domsearch/files/embassy-domsearch-0.1.0-r5.patch
	sci-biology/embassy-emnu/files/embassy-emnu-1.05-r7.patch
	sci-biology/embassy-esim4/files/embassy-esim4-1.0.0-r7.patch
	sci-biology/embassy-hmmer/files/embassy-hmmer-2.3.2-r4.patch
	sci-biology/embassy-iprscan/files/embassy-iprscan-4.3.1-r2.patch
	sci-biology/embassy-memenew/
	sci-biology/embassy-mira/
	sci-biology/embassy-mse/files/embassy-mse-3.0.0.patch
	sci-biology/embassy-phylipnew/files/embassy-phylipnew-3.69.patch
	sci-biology/embassy-signature/files/embassy-signature-0.1.0-r5.patch
	sci-biology/embassy-structure/files/embassy-structure-0.1.0-r5.patch
	sci-biology/embassy-topo/files/embassy-topo-2.0.0.patch
	sci-biology/embassy-vienna/files/embassy-vienna-1.7.2-r2.patch
	sci-biology/emboss/files/ax_lib_mysql.m4
	sci-biology/emboss/files/emboss-6.3.1_p4-r2_Makefile.patch
	sci-biology/emboss/files/emboss-6.3.1_p4-r2_compilations-paths.patch
	sci-biology/emboss/files/emboss-6.3.1_p4-r2_libtool.patch
	sci-biology/emboss/files/emboss-6.3.1_p4-r2_plcol.patch
	sci-biology/emboss/files/emboss-6.3.1_p4-r2_rename_plwid_to_plwidth.patch
	sci-biology/emboss/files/emboss-6.6.0_plcol.patch
	sci-biology/emboss/files/emboss-README.Gentoo-2
	sci-biology/spm/spm-8.ebuild.orig
	sci-libs/nipype/]

nothing added to commit but untracked files present 
```

**Comment by junghans** [comment]
```
chymera@zenbookhost ~/src/neurogentoo $ git commit -a
```
Why? Haven't we established already that the only way to add packages in by using `repoman commit`? Go into a package directory
```
cd sci-biology/spm
```
implement your changes
```
repoman manifest
repoman full
repoman -m "initial commit" commit
```
Don't forget to force push the `spm-octave` branch to github:
```
git push -f git@github.com:TheChymera/gentoo-overlay.git spm-octave:spm-octave
```

**Comment by TheChymera** [comment]
@junghans so what metadata is missing exactly?

```
chymera@zenbookhost ~/src/neurogentoo/sci-biology/spm $ repoman full
!!! Section 'neurogentoo' in repos.conf has name different from repository name 'science' set inside repository

RepoMan scours the neighborhood...
  ebuild.minorsyn               1
   sci-biology/spm/spm-8.ebuild: Useless blank line on line: 22
  metadata.missing [fatal]      1
   sci-biology/spm/metadata.xml
  upstream.workaround           3
   sci-biology/spm/spm-8.ebuild: Upstream parallel compilation bug (ebuild calls emake -j1 on line: 32)
   sci-biology/spm/spm-8.ebuild: Upstream parallel compilation bug (ebuild calls emake -j1 on line: 37)
   sci-biology/spm/spm-8.ebuild: Upstream parallel compilation bug (ebuild calls emake -j1 on line: 38)

Note: use --include-dev (-d) to check dependencies for 'dev' profiles

Please fix these important QA issues first.
RepoMan sez: "Make your QA payment on time and you'll never see the likes of me."
```

**Comment by jlec** [comment]
@TheChymera please try to gather some information yourself. Look at the repoman output or travis and do some research on what is going wrong here.

    ebuild.minorsyn               1
       sci-biology/spm/spm-8.ebuild: Useless blank line on line: 22

means useless blank line in 22 -> remove it.

    metadata.missing [fatal]      1
       sci-biology/spm/metadata.xml

no metadata.xml present -> google for "gentoo metadata.xml" to make yourself familiar with the topic. We are really spending lots of time helping you, but it seems your are not doing any effort to get information on your own. We are happy to help you, when you have unsolvable problems, but for such simple things I would kindly ask you to show some initiative.

**Comment by TheChymera** [comment]
Is this good now?

@jlec , yes, thank you for helping me, and I hope soon enough I will be able to thank your for pulling the ebuilds ;)

File: sci-biology/spm/spm-8.ebuild.orig [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Copyright 1999-2015 Gentoo Foundation
```
Comment (junghans) [inline sci-biology/spm/spm-8.ebuild.orig]: That file should not be in the commit!

File: sci-biology/spm/spm-8.ebuild [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+S=${WORKDIR}/${PN}${PV}
+
+src_prepare() {
+	echo "MEXOPTS += -v" >> src/Makefile.var
+	emake -C src distclean PLATFORM=octave
+	emake -C src toolbox-distclean PLATFORM=octave
+}
+
+src_compile() {
+	emake -C src -j1 PLATFORM=octave
```
Comment (junghans) [inline sci-biology/spm/spm-8.ebuild]: Question, have you tested that the `-j1` is really necessary?

File: sci-biology/spm/ChangeLog [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# ChangeLog for sci-biology/spm
+# Copyright 1999-2015 Gentoo Foundation; Distributed under the GPL v2
+# $Header: $
+
+  17 Feb 2015;  <chymera@gentoo.org> +metadata.xml, +spm-8.ebuild.orig,
```
Comment (junghans) [inline sci-biology/spm/ChangeLog]: For some reason your name doesn't show and your email seems to be wrong!
```
export ECHANGELOG_USER="First Last <123@456.org>"
```
might help!

File: sci-biology/spm/spm-8.ebuild [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+KEYWORDS="~amd64"
+
+RDEPEND=">=sci-mathematics/octave-3.6.4
+	"
+
+DEPEND="${RDEPEND}
+	app-arch/unzip
+	"
+
+S=${WORKDIR}/${PN}${PV}
```
Comment (junghans) [inline sci-biology/spm/spm-8.ebuild]: Please quote `S="${WORKDIR}/${PN}${PV}".

File: sci-biology/spm/spm-8.ebuild (hunk 4 above)
Comment (TheChymera) [inline sci-biology/spm/spm-8.ebuild]: how would I test that? I remember back one and a half years ago when I started to work on this, it failed because some parts were compiled ahead of other stuff they needed. Since the upstream code ha not changed afaict, this should have stayed the same.

File: sci-biology/spm/ChangeLog (hunk 5 above)
Comment (TheChymera) [inline sci-biology/spm/ChangeLog]: and then what? won't that set the variable just for my session? I this is an other repoman issue? where does repoman take my name and email from?

Perhaps we could also just let this rest? I don't mind if I am not credited for this particular thing. I feel we're getting too sidetracked by metadata....

File: sci-biology/spm/spm-8.ebuild (hunk 4 above)
Comment (junghans) [inline sci-biology/spm/spm-8.ebuild]: If you tested it earlier, then it is fine. I was just wondering if the `-j1` is actually needed.

File: sci-biology/spm/ChangeLog (hunk 5 above)
Comment (junghans) [inline sci-biology/spm/ChangeLog]: It is not about you getting credit, but for the users to know who break their system after installing a certain package. See
```
emerge -u --changelog --pretend world
```

Comment (TheChymera) [inline sci-biology/spm/ChangeLog]: Ok, so if I do decide to use `export`... as a temporary f what do I have to do after I run that? just `repoman commit` again? Also, where does repoman get my user data from? I heard it should be either `~/.gitconfig` or `./.git/config`. I guess the latter should not be cluttered with my user info, but the former states everything correctly. So what can I do to permanently tell repoman who I am?

chymera@zenbookhost ~/src/neurogentoo $ cat .git/config 
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = https://github.com/TheChymera/neurogentoo.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[branch "spm-octave"]
	remote = origin
	merge = refs/heads/spm-octave
[branch "patch-1"]
	remote = origin
	merge = refs/heads/patch-1
[branch "patch-2"]
	remote = origin
	merge = refs/heads/patch-2
[branch "fsl"]
	remote = origin
	merge = refs/heads/fsl
[branch "afni"]
	remote = origin
	merge = refs/heads/afni
[branch "pythontex-bump"]
	remote = origin
	merge = refs/heads/pythontex-bump
[remote "official"]
	url = https://github.com/gentoo-science/sci.git
	fetch = +refs/heads/*:refs/remotes/official/*
chymera@zenbookhost ~/src/neurogentoo $ cat ~/.gitconfig 
[user]
	email = h.chr@mail.ru
	name = Horea Christian
	signingkey = C7DE3E5F
[credential]
	helper = cache --timeout=99999
[push]
	default = simple
chymera@zenbookhost ~/src/neurogentoo $ cat sci-biology/spm/ChangeLog 
# ChangeLog for sci-biology/spm
# Copyright 1999-2015 Gentoo Foundation; Distributed under the GPL v2
# $Header: $

  17 Feb 2015;  <chymera@gentoo.org> -spm-8.ebuild.orig, spm-8.ebuild:
  added quotation marks, removed ori file

  17 Feb 2015;  <chymera@gentoo.org> +metadata.xml, +spm-8.ebuild.orig,
  spm-8.ebuild:
  added metadata
```

Comment (junghans) [inline sci-biology/spm/ChangeLog]: `repoman` is calling `echangelog` (see `man repoman` for details) and `echangelog` uses the perl function `getpwuid`, which uses the account information from your system (see `man echangelog` for details). The `ECHANGELOG_USER` variable is just the easiest way to overwrite your, maybe wrong, system settings. There is basically no harm in adding it to your `~/.bashrc`, but you could also export it and later unset it again.

And yeah, `echangelog` could use the information for your git settings, but @agriffis has implemented that back in 2009.

Comment (agriffis) [inline sci-biology/spm/ChangeLog]: *cough* 2003

File: sci-biology/spm/ChangeLog [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# ChangeLog for sci-biology/spm
+# Copyright 1999-2015 Gentoo Foundation; Distributed under the GPL v2
+# $Header: $
+
+  17 Feb 2015;  <chymera@gentoo.org> -spm-8.ebuild.orig, spm-8.ebuild:
```
Comment (junghans) [inline sci-biology/spm/ChangeLog]: Still wrong email here!

**Comment by TheChymera** [comment]
Ok, I removed the slots part, and I decided to just version this. Is this ok now, can you pull it?

**Comment by TheChymera** [comment]
@junghans bump?

**Comment by junghans** [comment]
Travis says:
```
   sci-biology/spm/spm-8.ebuild: Upstream parallel compilation bug (ebuild calls emake -j1 on line: 31)
   sci-biology/spm/spm-8.ebuild: Upstream parallel compilation bug (ebuild calls emake -j1 on line: 36)
   sci-biology/spm/spm-8.ebuild: Upstream parallel compilation bug (ebuild calls emake -j1 on line: 37)
   sci-biology/spm/spm-12.ebuild: Upstream parallel compilation bug (ebuild calls emake -j1 on line: 29)
```
but otherwise it looks good to me!
